src/libexpr/include/nix/expr/value.hh

```diff
@@ -17,25 +19,26 @@ namespace nix {
 struct Value;
 class BindingsBuilder;
 
-
 typedef enum {
```

Comment:
```suggestion
/**
 * Internal type discriminator, which is more detailed than `ValueType`, as
 * it specifies the exact representation used (for types that have multiple
 * possible representations).
 */
// The ordering is very significant. See Value::getInternalType() for details
// about how this is mapped into the alignment bits to save significant memory.
// This also restricts the number of internal types in each category.
typedef enum {
```

Reply:
Actually "category" here and in the comments below is not great. I don't think it's been explicitly named in the code yet, but I'm not entirely sure.
Other ideas:
- `shape`
- `layout`

This term can then also be mirrored in the setters, e.g: `setPairOfPointersPayload` -> `setPairOfPointersShapePayload`.